### ISSUE TITLE
Accessibility Documentation Alignment with Content Guidelines

### ISSUE THREAD

**Issue by jake-costa** [jpmorganchase/salt-ds#5465]
### Related page(s)

https://www.saltdesignsystem.com/salt/index

### Kind of issue

Missing information

### Description

Update component accessibility documentation to align with content guidelines feedback

### Browser

- [x] Chrome
- [x] Safari
- [x] Firefox
- [ ] Microsoft Edge

### Operating system

- [x] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.
